subsys/usb/usb_c/usbc_tc_src_states.c

```diff
@@ -295,6 +295,7 @@ void tc_attached_src_exit(void *obj)
 	tc_pd_enable(dev, false);
 
 	/* Stop sourcing VBUS */
+	__ASSERT(data->policy_cb_src_en != NULL, "Callback pointer should not be NULL");
```

Comment:
This assertion is already done on line 291. Maybe you could do something like the following:
```
int policy_cb_src_en(dev, bool en)
{
    struct usbc_port_data *data = dev->data;

     __ASSERT(data->policy_cb_src_en != NULL, "policy_cb_src_en must not be NULL");

    return data->policy_cb_src_en(dev, en);
 }
````
And just use the new function for VBUS control.

Reply:
We talked internally and decided that this callback should be made optional, since it can be unused when the TCPC/PPC controls the power paths. So asserts will be removed and ifs with checks added instead.